src/c-writer.cc

```diff
@@ -1158,10 +1202,11 @@ void CWriter::BeginInstance() {
         break;
       }
 
-      case ExternalKind::Table: {
-        Write("wasm_rt_table_t");
+      case ExternalKind::Table:
+        Write("wasm_rt_",
+              GetReferenceTypeName(cast<TableImport>(import)->table.elem_type),
+              "_table_t");
```

Comment:
Can we add a helper called `WriteTableType` that takes a type name? Then we could use it in `WriteTable` and `WriteTablePtr` below too?

Reply:
Sure, done.